File: rabbitmq/src/test/groovy/io/micronaut/configuration/rabbitmq/docs/quickstart/ProductListener.java
package io.micronaut.configuration.rabbitmq.docs.quickstart;

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Queue;
import io.micronaut.configuration.rabbitmq.annotation.RabbitListener;
import io.micronaut.context.annotation.Requires;

import java.util.ArrayList;
import java.util.List;
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartSpec")
// tag::class[]
@RabbitListener // <1>
public class ProductListener {

    public List<Integer> messageLengths = new ArrayList<>();

    @Queue("product") // <2>
    void receive(byte[] data) { // <3>
        Integer length = data.length;
        messageLengths.add(length);
        System.out.println("Received " + length + " bytes from RabbitMQ");
    }
}
// end::class[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/quickstart/ProductListener.kt
package io.micronaut.rabbitmq.docs.quickstart

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Queue
import io.micronaut.rabbitmq.annotation.RabbitListener
import java.util.Collections
// end::imports[]

@Requires(property = "spec.name", value = "QuickstartSpec")
// tag::clazz[]
@RabbitListener // <1>
class ProductListener {

    val messageLengths: MutableList<String> = Collections.synchronizedList(ArrayList())

    @Queue("product") // <2>
    fun receive(data: ByteArray) { // <3>
        val string = String(data)
        messageLengths.add(string)
        println("Kotlin received ${data.size} bytes from RabbitMQ: ${string}")
    }
}
// end::clazz[]
